### PR TITLE
Cvss Calc: Parse select values from url

### PR DESCRIPTION
**Pull request by saberlynx**
**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

**Comment by bjoernricks** [comment]
@saberlynx could you do me a favor and add some tests for this page too?

**Comment by saberlynx** [comment]
> @saberlynx could you do me a favor and add some tests for this page too?

Sure :)

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+const location = {
+  query: {cvssVector: 'AV:N/AC:L/Au:N/C:P/I:P/A:P'},
+};
+
+describe('CvssCalculator page tests', () => {
+  test('Should render with default values', () => {
+    const {render} = rendererWith({
+      gmp,
+      onInteraction,
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      expect(input[0]).toHaveTextContent('Local');
+      expect(input[1]).toHaveTextContent('Low');
+      expect(input[2]).toHaveTextContent('None');
+      expect(input[3]).toHaveTextContent('None');
+      expect(input[4]).toHaveTextContent('None');
+      expect(input[5]).toHaveTextContent('None');
+
+      expect(vector[0]).toHaveAttribute('value', 'AV:L/AC:L/Au:N/C:N/I:N/A:N');
+    });
+  });
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
  });

```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  test('Should render userVector from url', () => {
+    const {render} = rendererWith({
+      gmp,
+      onInteraction,
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      expect(input[0]).toHaveTextContent('Network');
+      expect(input[1]).toHaveTextContent('Low');
+      expect(input[2]).toHaveTextContent('None');
+      expect(input[3]).toHaveTextContent('Partial');
+      expect(input[4]).toHaveTextContent('Partial');
+      expect(input[5]).toHaveTextContent('Partial');
+
+      expect(vector[0]).toHaveAttribute('value', 'AV:N/AC:L/Au:N/C:P/I:P/A:P');
+    });
+  });
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
  });

```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  test('Changing userVector should change displayed select values', () => {
+    const {render} = rendererWith({
+      gmp,
+      onInteraction,
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+      expect(input[0]).toHaveTextContent('Network');
+      expect(input[1]).toHaveTextContent('Low');
+      expect(input[2]).toHaveTextContent('None');
+      expect(input[3]).toHaveTextContent('None');
+      expect(input[4]).toHaveTextContent('Partial');
+      expect(input[5]).toHaveTextContent('Partial');
+
+      expect(vector[0]).toHaveAttribute('value', 'AV:N/AC:L/Au:N/C:N/I:P/A:P');
+    });
+  });
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
  });

```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  test('Changing displayed select values should change userVector', async () => {
+    const {render} = rendererWith({
+      gmp,
+      onInteraction,
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
```

File: gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+describe('CvssCalculator page tests', () => {
+  test('Should render with default values', () => {
+    const {render} = rendererWith({
+      gmp,
+      store: true,
+    });
+
+    const {element, getAllByTestId} = render(<CvssCalculator />);
+
+    expect(element).toMatchSnapshot();
```
Comment (bjoernricks) [inline gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]:
```suggestion
```

Comment (saberlynx) [reply gsa/src/web/pages/extras/__tests__/cvsscalculatorpage.js]: Done :) And removed snapshot + its folder

**Pull request merged**